tcpdf.php

```diff
@@ -7866,6 +7891,52 @@ public function _destroy($destroyall=false, $preserve_objcopy=false) {
 		}
 	}
 
+	/** Page cache reference counts */
+	protected static $pageCacheRefCnts = [];
+
+	/**
+	 * Update page cache file
+	 *
+	 * @param int $memSizeInBytes Memory size to use in bytes
+	 * @return void
+	 */
+	public function usePageCacheFile($memSizeInBytes)
+	{
+		$this->pageCacheFile = fopen('php://temp/maxmemory:' . $memSizeInBytes, 'w+');
+		if ($this->pageCacheFile === false) {
+			$this->pageCacheFile = null;
+		} else {
+			self::$pageCacheRefCnts[(int)$this->pageCacheFile] = 1;
+		}
+	}
+
+	/**
+	 * Close page cache file
+	 */
+	private function _closePageCacheFile()
+	{
+		if ($this->pageCacheFile !== null)
+		{
+			self::$pageCacheRefCnts[(int)$this->pageCacheFile]--;
+			if (self::$pageCacheRefCnts[(int)$this->pageCacheFile] == 0) {
+				@fclose($this->pageCacheFile); // Suppress error since this might be the end of the PHP script
+			}
+			$this->pageCacheFile = null;
+			$this->pageCacheIndex = [];
+		}
+	}
+
+	/** Handle cloning */
```

Comment:
Why is this needed, it feels a bit incomplete or non explained

Reply:
I updated the comment.  It's so that if you clone the object, you don't have two objects writing (and overwriting each other) to the same file.